### PR TITLE
Add merging logic for V2Node

### PR DESCRIPTION
**Pull request by Spaceenter**

### CODE REVIEW

File: internal/server/handler_v2.go [hunk 1]
```diff
@@ -93,62 +91,57 @@ func (s *Server) V2Resolve(
...
-				s.store,
-				s.cache,
-				in.GetNodes(),
-				arc.SingleProp,
-				direction,
-				arc.Filter,
-			)
+		// Call remote Mixer and merge only for non-paginated cases,
+		// which are signaled by empty resp.GetNextToken().
+		if resp.GetNextToken() == "" && s.metadata.RemoteMixerURL != "" {
```
Comment (shifucun) [inline internal/server/handler_v2.go:109]: what if resp.GetNextToken() != ""?

Comment (shifucun) [reply internal/server/handler_v2.go:109]: I don't quite get the comment from line 107-108, maybe elaborate?

**Comment by shifucun** [comment]
Codacy check also suggests a branch simplification

File: internal/server/v2/propertyvalues/api.go [hunk 2]
```diff
@@ -83,6 +85,20 @@ func API(
 			}
 		}
 	}
+
+	var (
+		reqTokenMsg *v1.PaginationInfo
+		err         error
+	)
+	if reqToken == "" {
```
[the remark this answers is not in the document]
Comment (Spaceenter) [reply internal/server/v2/propertyvalues/api.go]: Updated with new recursive design.

File: internal/server/handler_v2.go (hunk 1 above)
Comment (Spaceenter) [reply internal/server/handler_v2.go]: Updated with new recursive design.

File: internal/merger/merger.go [hunk 3]
```diff
@@ -66,6 +68,47 @@ func MergeResolve(r1, r2 *pbv2.ResolveResponse) *pbv2.ResolveResponse {
...
+	res := &pbv2.NodeResponse{}
+
+	// Merge |data|.
+	res.Data = make(map[string]*pbv2.LinkedGraph)
+	for dcid, linkedGrarph := range local.GetData() {
+		res.Data[dcid] = linkedGrarph
+	}
+	for dcid, linkedGrarph := range remote.GetData() {
+		res.Data[dcid] = linkedGrarph
+	}
```
Comment (shifucun) [inline internal/merger/merger.go:83]: Should here merge within linkedGraph?

File: internal/merger/merger_test.go [hunk 4]
```diff
@@ -78,6 +79,155 @@ func TestMergeResolve(t *testing.T) {
...
+									{Value: "val1"},
+								},
+							},
+						},
+					},
+				},
+			},
+			&pbv2.NodeResponse{
+				Data: map[string]*pbv2.LinkedGraph{
+					"dcid2": {
```
Comment (shifucun) [inline internal/merger/merger_test.go:111]: maybe also test adding dcid1 in remote and combine nodes from local and remote?

File: internal/server/handler_v2.go [hunk 5]
```diff
@@ -93,65 +91,69 @@ func (s *Server) V2Resolve(
...
+		// 1. The call does not need pagination, e.g. PropertyLabels.
+		// 2. The call needs pagination, but this is the first call/page.
+		// In both cases, we need to read from both local and remote, and merge.
+
+		localResp, err := V2NodeCore(ctx, s.store, s.cache, s.metadata, in)
+		if err != nil {
+			return nil, err
+		}
+
+		if s.metadata.RemoteMixerDomain != "" {
```
Comment (shifucun) [inline internal/server/handler_v2.go:105]: maybe a TODO that these two calls can be done concurrently.

File: internal/server/v2/propertyvalues/golden/simple/california.json [hunk 6]
```diff
@@ -44,5 +44,6 @@
         }
       }
     }
-  }
+  },
+  "nextToken": "H4sIAAAAAAAA/wAAAP//AQAA//8AAAAAAAAAAA=="
```
Comment (shifucun) [inline internal/server/v2/propertyvalues/golden/simple/california.json:48]: should the nextToken be empty now?

**Review by shifucun — COMMENTED** [review]
looks very nice!

File: internal/server/v2/propertyvalues/golden/simple/california.json (hunk 6 above)
Comment (Spaceenter) [reply internal/server/v2/propertyvalues/golden/simple/california.json]: sorry forgot to update golden - now updated, and they're back to empty.

File: internal/server/handler_v2.go (hunk 5 above)
Comment (Spaceenter) [reply internal/server/handler_v2.go:105]: Added a TODO in the top of the file, as this applies to all V2 APIs.

File: internal/merger/merger_test.go (hunk 4 above)
Comment (Spaceenter) [reply internal/merger/merger_test.go:120]: done

File: internal/merger/merger.go (hunk 3 above)
Comment (Spaceenter) [reply internal/merger/merger.go]: done

**Review by shifucun — APPROVED** [review]
thx for the many rounds of change, this looks fantastic!

**Pull request merged**